Mathlib/Algebra/Group/Even.lean

```diff
@@ -91,63 +90,68 @@ lemma IsSquare.one [MulOneClass α] : IsSquare (1 : α) := ⟨1, (mul_one _).sym
 
 @[to_additive, deprecated (since := "2024-12-27")] alias isSquare_one := IsSquare.one
 
+section MonoidHom
+variable [MulOneClass α] [MulOneClass β] [FunLike F α β] [MonoidHomClass F α β]
+
 @[to_additive]
-lemma IsSquare.map [MulOneClass α] [MulOneClass β] [FunLike F α β] [MonoidHomClass F α β]
-    {m : α} (f : F) :
-    IsSquare m → IsSquare (f m) := by
-  rintro ⟨m, rfl⟩
-  exact ⟨f m, by simp⟩
+lemma IsSquare.map {a : α} (f : F) : IsSquare a → IsSquare (f a) :=
+  fun ⟨r, _⟩ => ⟨f r, by simp_all⟩
+
+end MonoidHom
 
 section Monoid
 variable [Monoid α] {n : ℕ} {a : α}
 
 @[to_additive even_iff_exists_two_nsmul]
-lemma isSquare_iff_exists_sq (m : α) : IsSquare m ↔ ∃ c, m = c ^ 2 := by simp [IsSquare, pow_two]
+lemma isSquare_iff_exists_sq (a : α): IsSquare a ↔ ∃ r, a = r ^ 2 := by simp [IsSquare, pow_two]
 
-alias ⟨IsSquare.exists_sq, isSquare_of_exists_sq⟩ := isSquare_iff_exists_sq
+@[to_additive Even.exists_two_nsmul
+  "Alias of the forwards direction of `even_iff_exists_two_nsmul`."]
+alias ⟨IsSquare.exists_sq, _⟩ := isSquare_iff_exists_sq
 
-attribute [to_additive Even.exists_two_nsmul
-  "Alias of the forwards direction of `even_iff_exists_two_nsmul`."] IsSquare.exists_sq
+@[to_additive (attr := simp) Even.two_nsmul]
+lemma IsSquare.sq (r : α) : IsSquare (r ^ 2) := ⟨r, pow_two _⟩
+
+@[deprecated (since := "2024-12-27")] alias even_two_nsmul := Even.two_nsmul
+@[deprecated (since := "2024-12-27")] alias IsSquare_sq := IsSquare.sq
```

Comment:
This might help untangle the diff
```suggestion
@[deprecated (since := "2024-12-27")] alias IsSquare_sq := IsSquare.sq
@[deprecated (since := "2024-12-27")] alias even_two_nsmul := Even.two_nsmul
```